### PR TITLE
GH1403: Add optional Go.CD Server URL Parameter to GetHistory

### PR DESCRIPTION
**Pull request by RehanSaeed**
A small tweak to the Go.CD GetHistory function to allow you to optionally set a GO.CD server URL.

### CODE REVIEW

**Comment by gep13** [comment]
@RehanSaeed thank you for submitting this change.  Before going any further, can I please ask you to read our [contributing guide](https://github.com/cake-build/cake/blob/develop/CONTRIBUTING.md) and please take the necessary steps.

**Comment by RehanSaeed** [comment]
@gep13 Was in the middle of writing the issue https://github.com/cake-build/cake/issues/1403

**Comment by gep13** [comment]
>@RehanSaeed said...
@gep13 Was in the middle of writing the issue #1403 which I assume is what you are referring to.

That's the one, yes.  That guideline is put in place to protect contributors.  In this case, it isn't/wasn't really needed, but we prefer to get sign off about any chances, prior to any work being done.  That way, there is less chance of a PR not being accepted.

File: src/Cake.Common/Build/GoCD/IGoCDProvider.cs [hunk 1]
```diff
@@ -32,7 +32,8 @@ public interface IGoCDProvider
         /// </summary>
         /// <param name="username">The Go.CD username.</param>
         /// <param name="password">The Go.CD password.</param>
+        /// <param name="serverUrl">The Go.CD server url.</param>
         /// <returns>The Go.CD build history.</returns>
-        GoCDHistoryInfo GetHistory(string username, string password);
+        GoCDHistoryInfo GetHistory(string username, string password, string serverUrl = null);
```
Comment (patriksvensson) [inline src/Cake.Common/Build/GoCD/IGoCDProvider.cs]: Would prefer to have two overloads here instead of an optional argument.

Comment (devlead) [inline src/Cake.Common/Build/GoCD/IGoCDProvider.cs]: Yes, optional parameters breaks ABI, so overload is needed.

**Comment by RehanSaeed** [comment]
Switched to two overloaded methods.

**Comment by devlead** [comment]
@RehanSaeed your changes have been merged, thanks for contributing 👍

**Pull request merged**